### PR TITLE
fix: also delete parserOptions.EXPERIMENTAL_useProjectService

### LINKED ISSUE #2962: When clearing typed linting options, also delete parserOptions.EXPERIMENTAL_useProjectService
Following https://github.com/typescript-eslint/typescript-eslint/issues/8424#issuecomment-1937117103: right now, `eslint-plugin-import` deletes the two known `parserOptions` from [typescript-eslint's typed linting](https://typescript-eslint.io/linting/typed-linting):

https://github.com/import-js/eslint-plugin-import/blob/7a21f7e10f18c04473faadca94928af6b8e28009/utils%2Fparse.js#L80-L85

...but we added an `EXPERIMENTAL_useProjectService` too. Folks using it are experiencing the performance issues from #1408 again.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Fixes #2962.

### CODE REVIEW

File: utils/parse.js [hunk 1]
```diff
@@ -81,6 +81,7 @@ exports.default = function parse(path, content, context) {
   // "project" or "projects" in parserOptions. Removing these options means the parser will
   // only parse one file in isolate mode, which is much, much faster.
   // https://github.com/import-js/eslint-plugin-import/issues/1408#issuecomment-509298962
+  delete parserOptions.EXPERIMENTAL_useProjectService;
   delete parserOptions.project;
   delete parserOptions.projects;
```
Comment (JoshuaKGoldberg) [inline utils/parse.js:86]: We'll eventually add a new name for the option like `projectService` or something like it. Which would increase this list to four properties.

An alternative to hardcoding names in this plugin would be https://github.com/typescript-eslint/typescript-eslint/issues/8428: importing a shared utility we provide. Is that something you'd find convenient? Either as a part of one of our packages _(easiest for us)_ or as a standalone thing _(we'd have to discuss)_?

Comment (JounQin) [reply utils/parse.js:86]: It's late to add a util for compatibility, what means deleting manually is required for old `@typescript-eslint` versions.

**Review by JounQin — APPROVED** [review]

File: utils/parse.js (hunk 1 above)
Comment (ljharb) [reply utils/parse.js:86]: For new ones, though, a conditional require would be just fine.

Comment (ljharb) [reply utils/parse.js:86]: sure, sounds good

**Review by ljharb — APPROVED** [review]

**Pull request merged**